_posts/2021-02-02-getting-started-amqp.adoc

```diff
@@ -0,0 +1,335 @@
+@Produces(MediaType.APPLICATION_JSON)
+public class MovieResource {
+
+    @Inject
+    MovieProducer producer;
+
+    @POST
+    public Response send(Movie movie) {
+        producer.send(movie);
+        // Return an 202 - Accepted response.
+        return Response.accepted().build();
+    }
+}
+----
+
+This class uses the `MovieProducer` we implemented above to send the `movies`.
+You could also use an `Emitter` directly.
+
+== Step 7 - Let's get this running!
+
+Well, first, we need an AMQP broker.
+You can follow the http://activemq.apache.org/components/artemis/documentation/1.0.0/running-server.html[Getting Started with Artemis] documentation, or use the following `docker-compose.yaml` file:
+
+[source, yaml]
+----
+version: '2'
+
+services:
+
+  artemis:
+    image: vromero/activemq-artemis:2-alpine-latest
+    ports:
+      - "8161:8161"
+      - "61616:61616"
+      - "5672:5672"
+    environment:
+      ARTEMIS_USERNAME: quarkus
+      ARTEMIS_PASSWORD: quarkus
+----
+
+Copy the `docker-compose.yaml` file in your project, and from a terminal, start your broker with: `docker-compose up -d'
+
+Then, run the application using:
+
+[source, bash]
+----
+./mvnw quarkus:dev
+----
+
+The application runs in dev mode, meaning that you can still update the code.
+It will reload it automatically.
+
+In another terminal, emit a few HTTP POST request such as:
+
+[source, bash]
+----
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1994, "title":"The Shawshank Redemption"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1972, "title":"The Godfather"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":2008, "title":"The Dark Knight"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1994, "title":"Pulp Fiction"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":2010, "title":"Inception"}' \
+http://localhost:8080/
+----
+
+In the terminal running the application, you will see:
+
+[source, text]
+----
+...
+2021-01-27 09:29:41,087 INFO  [org.acm.MovieConsumer] (vert.x-eventloop-thread-9) Got a movie: 1994 - Pulp Fiction
+2021-01-27 09:29:41,114 INFO  [org.acm.MovieConsumer] (vert.x-eventloop-thread-9) Got a movie: 2010 - Inception
+...
+----
+
+It works!
+
+== Step 8 - Native packaging
+
+If you have https://quarkus.io/guides/building-native-image[GraalVM installed and configured correctly], you can package this application as a native executable:
+
+[source, bash]
+----
+./mvnw package -Pnative
+----
+
+Then, execute your native executable with: `./target/getting-started-amqp-1.0.0-SNAPSHOT-runner`, and you get a Quarkus application using AMQP starting in a few milliseconds and consuming a ridiculous amount of memory: 33Mb after 100 ingested records!
```

Comment:
Perhaps 'only 33Mb', otherwise it reads a bit more like thats a bad amount to use.

Reply:
Fixed